### PR TITLE
Edit previously submitted application

### PR DESCRIPTION
**Pull request by elahti**
User can edit previously submitted application by navigating to `/hakemus?modify=:secret`. This features *does not yet send the secret URL by email*.

### CODE REVIEW

File: src/clj/ataru/hakija/hakija_routes.clj [hunk 1]
```diff
@@ -59,6 +59,19 @@
         (info "Started person creation job (to person service) with job id" person-service-job-id)
         (response/ok {:id application-id})))))
 
+(defn- get-application [secret]
+  (let [application (application-store/get-latest-application-by-secret secret)]
+    (cond
+      (some? (:id application))
+      (do
+        (info (str "Getting application " (:id application) " with answers"))
+        (response/ok (dissoc application :secret)))
```
Comment (elahti) [inline src/clj/ataru/hakija/hakija_routes.clj]: It was accidentally left there. Removed.

**Comment by auramo** [comment]
When editing non-finnish applicant's application, the selection "Onko suomalainen henkilötunnus" is missing from the form. It is present when I am applying for the first time. It is missing for both birthDate and hetu-cases.


screen shot 2016-11-11 at 11 44 20

screen shot 2016-11-11 at 11 41 23

When trying to save modifications "Lähetä muutokset", this happens:

screen shot 2016-11-11 at 11 45 28

File: src/cljc/ataru/schema/form_schema.cljc [hunk 2]
```diff
@@ -144,7 +144,9 @@
    (s/optional-key :id)             s/Int
    (s/optional-key :hakukohde)      (s/maybe s/Str)
    (s/optional-key :hakukohde-name) (s/maybe s/Str)
-   (s/optional-key :created-time)   org.joda.time.DateTime})
+   (s/optional-key :created-time)   org.joda.time.DateTime
+   (s/optional-key :secret)         s/Str
```
Comment (starvinraven) [inline src/cljc/ataru/schema/form_schema.cljc]: Was this necessary, if the secret is never transmitted over the wire?

File: src/clj/ataru/applications/application_service.clj [hunk 3]
```diff
@@ -61,7 +63,7 @@
   "Get application that has human-readable koodisto values populated
    onto raw koodi values."
   [application-key session organization-service]
-  (let [application (application-store/get-latest-application-by-key application-key)
+  (let [application (dissoc (application-store/get-latest-application-by-key application-key) :secret)
```
Comment (starvinraven) [inline src/clj/ataru/applications/application_service.clj]: Looks like the dissoc is not necessary here?

**Pull request merged**